src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -613,6 +613,121 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### Defining custom hooks
+
+As an integration author, you can define custom hooks that your integration will call. This allows for runtime interactions between your integration and other integrations. Other integrations can provide implementations for your custom hooks and your integration can call those at the appropriate moment in your integration's lifecycle.
+
+#### Defining hook types
+
+Provide the type for your custom hook by augmenting Astro’s `IntegrationHooks` interface within your integration:
+
+```ts title="types.ts"
+// on an implementation file
+declare global {
+  namespace Astro {
+    interface IntegrationHooks {
+      'myLib:eventHappened': (your: string, parameters: number) => Promise<void>;
+    }
+  }
+}
+```
+
+#### Calling custom hooks
+
+Installed integrations can be accessed from the `astro:config:setup` and `astro:config:done` hooks. Loop over these to call your custom hook on each if available:
+
+```ts {7-9}
+// your-integration/index.ts
+export default (): AstroIntegration => {
+  return {
+    name: 'your-integration',
+    hooks: {
+      'astro:config:setup': async ({ config }) => {
+        for (const integration of config.integrations) {
+          await integration.hooks['myLib:eventHappened'].?('your values', 123);
+        }
+      },
+    }
+  }
+}
+```
+
+To call custom hooks outside of `astro:config:setup` and `astro:config:done`, store the integrations array and use it later. `astro:config:done` should be preferred to work with integrations that get added by other integrations during `astro:config:setup`.
+
+```ts {3,9-10,13-16}
+// your-integration/index.ts
+export default (): AstroIntegration => {
+  let integrations: AstroIntegration[] = [];
+
+  return {
+    name: 'your-integration',
+    hooks: {
+      'astro:config:done': async ({ config }) => {
+        // Store the integrations when available
+        integrations = config.integrations;
+      },
+      'astro:build:done': async () => {
+        // Loop over the stored integrations in a later hook
+        for (const integration of config.integrations) {
+          await integration.hooks['myLib:eventHappened'].?('your values', 123);
+        }
+      },
+    }
+  }
+}
+```
+
+#### Using a custom hook from another integration
```

Comment:
This doesn't immediately feel like the right description of what's described below, and I can't quite put my finger on it yet. This sounds like a thing *you* want to do (use someone else's integration), but the text below is about *someone else* using *your* hook. It's not wrong, but it doesn't *feel* exactly right.

I would be tempted to either change this heading to more imply "making your hook available to other integrations" to match the example below, or to change the example below to show YOU tapping into someone ELSE's custom hook. These feel like two different things.

Reply:
@Fryuni Did you have any particular thoughts here re: what an integration author might more often be thinking/their need or perspective when they read these docs?

Would you say it's more, "I need to use someone else's custom hook in their integration" or "I need to make my hook available for other people to use in their integration" ? Although practically speaking they may result in similar examples, I feel like these are two different use cases and would like the text to match the thing someone is trying to do. 

As I think I put in a comment below, if the *making your hook available* part is really like an extra step in your process when you create your hook, that could go earlier, then this section could really be about how to use someone else's hook.

Reply:
Originally, I wrote each section from the perspective of who I'd expect to use them. Chris suggested keeping the same perspective throughout all sections, so I changed the text and examples to be from the perspective of the integration author defining the hooks.